cmd/health/health.go

```diff
@@ -0,0 +1,192 @@
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+package health
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/aurae-runtime/ae/pkg/cli"
+	"github.com/aurae-runtime/ae/pkg/cli/printer"
+	"github.com/aurae-runtime/ae/pkg/client"
+	"github.com/aurae-runtime/ae/pkg/config"
+	"github.com/spf13/cobra"
+
+	aeCMD "github.com/aurae-runtime/ae/cmd"
+	discoveryv0 "github.com/aurae-runtime/ae/pkg/api/v0/discovery"
+)
+
+type outputHealthNode struct {
+	Version string `json:"version"`
+}
+
+type outputHealth struct {
+	Nodes map[string]outputHealthNode `json:"nodes"`
+}
+
+func inc(ip net.IP) {
+	for i := len(ip) - 1; i >= 0; i-- {
+		ip[i]++
+		if ip[i] > 0 {
+			break
+		}
+	}
+}
+
+func hosts(cidr string) ([]string, error) {
+	ip, ipnet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return nil, err
+	}
+
+	var ips []string
+	for ip := ip.Mask(ipnet.Mask); ipnet.Contains(ip); inc(ip) {
+		ips = append(ips, ip.String())
+	}
+	// remove network address and broadcast address
+	if strings.HasSuffix(ips[0], ".0") {
+		ips = ips[1:]
+	}
+	if strings.HasSuffix(ips[len(ips)-1], ".255") {
+		ips = ips[:len(ips)-1]
+	}
+	return ips, nil
+}
+
+func protocol(ip_str string) (string, error) {
```

Comment:
See my note, should we move this to a library or use another library? I assume the tailscale codebase should have some IPv4/IPv6 utilities we can probably leverage?